packages/ui/src/components/InputFields/PasswordInputField/index.tsx

```diff
@@ -19,21 +18,17 @@ const PasswordInputField = (props: Props, forwardRef: Ref<Nullable<HTMLInputElem
 
   useImperativeHandle(forwardRef, () => innerRef.current);
 
-  // Refocus and move cursor to the end of the input field after password visibility is toggled
-  useUpdateEffect(() => {
-    if (innerRef.current) {
-      const { length } = innerRef.current.value;
-      innerRef.current.focus();
-      innerRef.current.setSelectionRange(length, length);
-    }
-  }, [showPassword]);
-
   return (
     <InputField
       isSuffixFocusVisible
       type={showPassword ? 'text' : 'password'}
       suffix={
-        <IconButton onClick={toggleShowPassword}>
+        <IconButton
+          onMouseDown={(event) => {
+            event.preventDefault();
+            toggleShowPassword();
+          }}
+        >
```

Comment:
nit: It would be better to still handle the toggle on click, but leaving prevent default on mouse down

Reply:
updated